### PR TITLE
feat: docs deploy previews

### PR DESCRIPTION
**Pull request by BRKalow**
---
Blocked by https://github.com/hashicorp/terraform-website/pull/2209

### CODE REVIEW

**Comment by ansgarm** [comment]
@BRKalow is there a way to let Vercel only run on PRs that changed doc related files?
It seems a bit noisy on regular PRs 😅

**Comment by BRKalow** [comment]
@ansgarm Yeah great call! I've added the `github.silent` config, which we also have across our other product repos, as well as a step to only trigger a build if there are changes in the website directory.

PR: https://github.com/hashicorp/terraform-cdk/pull/1611

**Comment by ansgarm** [comment]
Hi @BRKalow 👋 Is this still WIP or ready to review?

**Comment by ansgarm** [comment]
@BRKalow fyi, the Vercel build seems to [fail on PRs](https://github.com/hashicorp/terraform-cdk/pull/1586#issuecomment-1074236962) that are opened from a fork.

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+	@docker run ${DOCKER_RUN_FLAGS} ${DOCKER_IMAGE} npm start
+
+website/local:
+	@echo "==> Starting website in Docker..."
+	@docker run ${DOCKER_RUN_FLAGS} ${DOCKER_IMAGE_LOCAL} npm start
+
+.PHONY: website/build-local
+website/build-local:
+	@echo "==> Building local Docker image"
+	@docker build https://github.com/hashicorp/terraform-website.git\#brk.feat/tfcdk-remote-content \
```
Comment (thiskevinwang) [inline Makefile:27]: TODO: update branch to `master` when https://github.com/hashicorp/terraform-website/pull/2209 is merged

Comment (thiskevinwang) [inline Makefile:27]:
```suggestion
	@docker build https://github.com/hashicorp/terraform-website.git\#master \
```

**Review by thiskevinwang — APPROVED** [review]
LGTM

**Review by ansgarm — APPROVED** [review]

**Comment by thiskevinwang** [comment]
@ansgarm FYI: I don't have perms to merge this so I might need your help in doing so

**Comment by ansgarm** [comment]
Done 🎉

**Pull request merged**